dataactcore/utils/duns.py

```diff
@@ -65,156 +64,58 @@ def get_client(ssh_key=None):
-#     del data["old_activation_date"]
-#     if benchmarks:
-#         logger.info("Activation check took {} seconds".format(time.time()-activation_check_start))
-#     return data
-
-def update_duns(models, new_data, benchmarks=False, table=DUNS):
-    """ Modify existing models or create new ones
-
-        Args:
-            models: the DUNS objects representing the updated data
-            new_data: the new data to update
-            benchmarks: whether or not to log times
-            table: the table to work from (could be DUNS/HistoricParentDuns)
-
-        Returns:
-            tuple of added and updated duns lists
-    """
-    added = []
-    updated = []
-    logger.info("Updating duns")
-    if benchmarks:
-        update_duns_start = time.time()
-    for _, row in new_data.iterrows():
-        awardee_or_recipient_uniqu = row['awardee_or_recipient_uniqu']
-        if awardee_or_recipient_uniqu not in models:
-            models[awardee_or_recipient_uniqu] = table()
-            added.append(awardee_or_recipient_uniqu)
-        else:
-            updated.append(awardee_or_recipient_uniqu)
-        for field, value in row.items():
-            if value:
-                setattr(models[awardee_or_recipient_uniqu], field, value)
-    if benchmarks:
-        logger.info("Updating duns took {} seconds".format(time.time() - update_duns_start))
-    return added, updated
-
-
-def clean_sam_data(data, table=DUNS):
+def clean_sam_data(data):
     """ Wrapper around clean_data with the DUNS context
 
         Args:
             data: the dataframe to be cleaned
-            table: the table to work from (could be DUNS/HistoricParentDuns)
 
         Returns:
             a cleaned/updated dataframe to be imported
     """
-    return clean_data(data, table, {
-        "awardee_or_recipient_uniqu": "awardee_or_recipient_uniqu",
-        "activation_date": "activation_date",
-        "deactivation_date": "deactivation_date",
-        "registration_date": "registration_date",
-        "expiration_date": "expiration_date",
-        "last_sam_mod_date": "last_sam_mod_date",
-        "sam_extract_code": "sam_extract_code",
-        "legal_business_name": "legal_business_name",
-        "dba_name": "dba_name",
-        "address_line_1": "address_line_1",
-        "address_line_2": "address_line_2",
-        "city": "city",
-        "state": "state",
-        "zip": "zip",
-        "zip4": "zip4",
-        "country_code": "country_code",
-        "congressional_district": "congressional_district",
-        "entity_structure": "entity_structure",
-        "business_types_codes": "business_types_codes",
-        "ultimate_parent_legal_enti": "ultimate_parent_legal_enti",
-        "ultimate_parent_unique_ide": "ultimate_parent_unique_ide"
-    }, {})
-
-
-def parse_duns_file(file_path, sess, monthly=False, benchmarks=False, table=DUNS, year=None, metrics=None):
+    if not data.empty:
+        return clean_data(data, DUNS, {
```

Comment:
Should this return None or something if the data IS empty?

Reply:
Good catch, this was added for a special test case where there we no delete records. It should return the same empty table if it's empty.